conans/client/toolchain/make.py

```diff
@@ -0,0 +1,229 @@
+# coding=utf-8
+
+import platform
+import textwrap
+
+from jinja2 import Template
+
+from conans.client.build.compiler_flags import architecture_flag, build_type_flags
+from conans.client.build.cppstd_flags import cppstd_from_settings, cppstd_flag_new as cppstd_flag
+from conans.client.tools.oss import cross_building, \
+    detected_architecture, detected_os, get_gnu_triplet, get_target_os_arch, get_build_os_arch
+from conans.errors import ConanException
+from conans.util.files import save
+
+
+class MakeToolchain(object):
+    filename = "conan_toolchain.mak"
+
+    _template_toolchain = textwrap.dedent("""
+        # Conan generated toolchain file
+        ifndef CONAN_TOOLCHAIN_INCLUDED
+            CONAN_TOOLCHAIN_INCLUDED = TRUE
+            CONAN_TC_BUILD_TYPE = {{build_type}}
```

Comment:
Why the ``CONAN_TC_XXX`` naming and not directly ``CONAN_XXX``?

Reply:
To clearly separate and avoid overlap or conflict with make generator.